### ISSUE TITLE
3.5.2 M3 Software Design Documentation

### ISSUE THREAD

**Issue by jasonyjk** [INF2001/TEAM-PROJECT-P11-3#127]

**Comment by jasonyjk** [comment]
1. Task title ID: **#127**

2. What is the task about? **To create a detailed blueprint of the software's architecture, components, and interaction patterns, aimed at guiding the development process and ensuring alignment with technical and business requirements.**
3. What are the goals of the task? **To provide a comprehensive and detailed description of the software's architecture and design, ensuring clarity in implementation and facilitating consistent, quality software development aligned with specified requirements and standards.**
4. Success or completion criteria: **Clear articulation of the MVC and layered architectures integration, along with a detailed exposition of the Strategy and Factory design patterns, demonstrating the system's scalability, security, and maintainability.**

5. Allocated time: **4 days (96 hours)**
6. Start date and end date of the issue/task **Start: 15/11/2023 End: 18/11/2023**
7. Who is the owner? **Jason**
8. The status of the task: **Completed**
9. What is the Priority of the task? **High***

**Comment by jasonyjk** [comment]
1. In our detailed work process, we have compiled and documented the Architecture Design section highlighting the hybrid design's strengths in modularity, scalability, code efficiency, security, and adaptability. In the Detailed Design section, we highlighted the implementation of Strategy and Factory Patterns that foster flexible decision-making, modular design, component reusability, extensibility, and consistent object creation within the software structure.
2. The work has been updated onto M3 Report, Section 2 Software Design, sub-section 2.1 Architecture Design and 2.2 Detailed Design (2.2.1 Strategy Pattern and 2.2.2 Factory Pattern).
3. The project timeline remains unaffected, and it continues to progress as per the established schedule.
4. No, there are no changes to the success or completion criteria of the task. The criteria remain as originally defined.
5. Start date 15/11/2023 end date 18/11/2023
